cmd/podman/kill.go

```diff
@@ -51,54 +50,45 @@ func init() {
 
 // killCmd kills one or more containers with a signal
 func killCmd(c *cliconfig.KillValues) error {
-	var (
-		killFuncs  []shared.ParallelWorkerInput
-		killSignal uint = uint(syscall.SIGTERM)
-	)
-
-	rootless.SetSkipStorageSetup(true)
-	runtime, err := libpodruntime.GetRuntime(&c.PodmanCommand)
-	if err != nil {
-		return errors.Wrapf(err, "could not get runtime")
+	if c.Bool("trace") {
+		span, _ := opentracing.StartSpanFromContext(Ctx, "killCmd")
+		defer span.Finish()
 	}
-	defer runtime.Shutdown(false)
 
-	if c.Signal != "" {
+	if c.Flags().Changed("signal") {
 		// Check if the signalString provided by the user is valid
 		// Invalid signals will return err
-		sysSignal, err := signal.ParseSignal(c.Signal)
-		if err != nil {
+		if _, err := signal.ParseSignal(c.Signal); err != nil {
 			return err
 		}
-		killSignal = uint(sysSignal)
 	}
 
-	containers, err := getAllOrLatestContainers(&c.PodmanCommand, runtime, libpod.ContainerStateRunning, "running")
+	rootless.SetSkipStorageSetup(true)
+	runtime, err := adapter.GetRuntime(&c.PodmanCommand)
 	if err != nil {
-		if len(containers) == 0 {
-			return err
-		}
-		fmt.Println(err.Error())
+		return errors.Wrapf(err, "could not get runtime")
 	}
+	defer runtime.Shutdown(false)
 
-	for _, ctr := range containers {
-		con := ctr
-		f := func() error {
-			return con.Kill(killSignal)
-		}
-
-		killFuncs = append(killFuncs, shared.ParallelWorkerInput{
-			ContainerID:  con.ID(),
-			ParallelFunc: f,
-		})
+	ok, failures, err := runtime.KillContainers(getContext(), c)
+	if err != nil {
+		return err
 	}
 
-	maxWorkers := shared.Parallelize("kill")
-	if c.GlobalIsSet("max-workers") {
-		maxWorkers = c.GlobalFlags.MaxWorks
+	for _, id := range ok {
+		fmt.Println(id)
 	}
-	logrus.Debugf("Setting maximum workers to %d", maxWorkers)
 
-	killErrors, errCount := shared.ParallelExecuteWorkerPool(maxWorkers, killFuncs)
-	return printParallelOutput(killErrors, errCount)
+	if len(failures) > 0 {
+		keys := reflect.ValueOf(failures).MapKeys()
```

Comment:
Oh dear... Are we using this because the type differs between remote and local?

Reply:
Building a slice containing the map's keys to support same output as previous code.